### PR TITLE
s22-4pm-2 Eren - add endpoint returns sections for psId

### LINKED ISSUE #26: Add endpoint that returns, for a personal schedule, a list of all of the enrollCd's of the sections in the database that are in that personal schedule.

### PR DESCRIPTION
**Pull request by E-R-E-N**
To close #26 
# Overview
Added endpoint that returns, for a personal schedule, a list of all of the enrollCd's of the sections in the database that are in that personal schedule.
Revised Courses entity & repository, Course, CourseInfo, CoursePage,UCSBCurriculumService for applying functions in SectionController.
Implemented tests for SectionController, CoursePage, and UCSBCurriculumService.
Passed all Cl/CD.

### CODE REVIEW

**Comment by mayachristinaha** [comment]
Please add more to your PR description

**Comment by E-R-E-N** [comment]
> 
Added content for the PR.

File: src/main/java/edu/ucsb/cs156/courses/documents/CourseInfo.java [hunk 1]
```diff
@@ -18,5 +19,20 @@ public class CourseInfo {
...
+    private int unitsVariableLow;
+    private boolean delayedSectioning;
+    private boolean inProgressCourse;
+    private String gradingOption;
+    private String instructionType;
+    private boolean onLineCourse;
+    private String deptCode;
+    private List<GeneralEducation> generalEducation;
+    private List<Section> classSections;
+}
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/cs156/courses/documents/CourseInfo.java]: The fields in this document are exactly the same as that of `Course.java`. Why do we have two documents with the exact same member fields?

File: src/test/java/edu/ucsb/cs156/courses/controllers/SectionControllerTests.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import edu.ucsb.cs156.courses.config.SecurityConfig;
+import edu.ucsb.cs156.courses.services.UCSBCurriculumService;
+import edu.ucsb.cs156.courses.entities.Courses;
+import edu.ucsb.cs156.courses.repositories.UserRepository;
+import edu.ucsb.cs156.courses.repositories.CoursesRepository;
+import edu.ucsb.cs156.courses.documents.ConvertedSection;
+import edu.ucsb.cs156.courses.documents.CoursePage;
+import edu.ucsb.cs156.courses.documents.CourseInfo;
+import edu.ucsb.cs156.courses.documents.CoursePageFixtures;
+import edu.ucsb.cs156.courses.ControllerTestCase;
```
Comment (andrewhlu) [inline src/test/java/edu/ucsb/cs156/courses/controllers/SectionControllerTests.java:12]: nit: can we alphabetize the imports?

File: src/test/java/edu/ucsb/cs156/courses/controllers/SectionControllerTests.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+@WebMvcTest(value = SectionController.class)
+@Import(SecurityConfig.class)
+public class SectionControllerTests extends ControllerTestCase {
+    private final Logger logger = LoggerFactory.getLogger(SectionControllerTests.class);
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @MockBean
+    CoursesRepository coursesRepository;
+
+     @MockBean
```
Comment (andrewhlu) [inline src/test/java/edu/ucsb/cs156/courses/controllers/SectionControllerTests.java:43]: this bothers me slightly ahaha
```suggestion
    @MockBean
```

**Review by andrewhlu — CHANGES_REQUESTED** [review]

File: src/test/java/edu/ucsb/cs156/courses/controllers/SectionControllerTests.java (hunk 2 above)
Comment (E-R-E-N) [reply src/test/java/edu/ucsb/cs156/courses/controllers/SectionControllerTests.java]: Now they are in alphabetical order.

File: src/main/java/edu/ucsb/cs156/courses/documents/CourseInfo.java (hunk 1 above)
Comment (E-R-E-N) [reply src/main/java/edu/ucsb/cs156/courses/documents/CourseInfo.java]: Because of the different formats of functions when calling these variables.

**Pull request closed without merging**